mmedit/datasets/pipelines/augmentation.py

```diff
@@ -1131,3 +1131,67 @@ def __call__(self, results):
 
     def __repr__(self):
         return self.__class__.__name__
+
+
+@PIPELINES.register_module()
+class UnsharpMasking:
+    """Apply unsharp masking to an image or a sequence of images.
+
+    Args:
+        kernel_size (int): The kernel_size of the Gaussian kernel.
+        sigma (float): The standard deviation of the Gaussian.
+        weight (float): The weight of the "details" in the final output.
+        threshold (float): Pixel differences larger than this value are
+            regarded as "details".
+        keys (list[str]): The keys whose values are processed.
+
+    Added keys are "xxx_unsharp", where "xxx" are the attributes specified
+    in "keys".
+
+    """
+
+    def __init__(self, kernel_size, sigma, weight, threshold, keys):
+        if kernel_size % 2 == 0:
+            kernel_size += 1
+
+        self.kernel_size = kernel_size
+        self.sigma = sigma
+        self.weight = weight
+        self.threshold = threshold
+        self.keys = keys
+
+        kernel = cv2.getGaussianKernel(kernel_size, sigma)
+        self.kernel = np.dot(kernel, kernel.transpose())
+
+    def _unsharp_masking(self, imgs):
+        is_single_image = False
+        if isinstance(imgs, np.ndarray):
+            is_single_image = True
+            imgs = [imgs]
+
+        outputs = []
+        for img in imgs:
+            residue = img - cv2.filter2D(img, -1, self.kernel)
```

Comment:
what's the type of img? if it is uint8, then is this `-` correct?

Reply:
In our pipeline, the images should be loaded as a float. I think it is fine?